org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/util/widget/GradleHoverHelp.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2017 the original author or authors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.buildship.ui.util.widget;
+
+import com.google.common.base.Preconditions;
+
+import org.eclipse.jface.text.AbstractInformationControl;
+import org.eclipse.jface.text.DefaultInformationControl;
+import org.eclipse.swt.events.MouseEvent;
+import org.eclipse.swt.events.MouseListener;
+import org.eclipse.swt.events.MouseMoveListener;
+import org.eclipse.swt.events.MouseTrackAdapter;
+import org.eclipse.swt.graphics.Point;
+import org.eclipse.swt.graphics.Rectangle;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.swt.widgets.Widget;
+
+/**
+ * Displays a hover message for the target UI control.
+ * <p/>
+ * The implementation is based on the HoverHelp example found in the <a href=
+ * "https://github.com/eclipse/eclipse.platform.swt/tree/master/examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/hoverhelp">eclipse/eclipse.plaform.swt</a>
+ * repository.
+ *
+ * @author Donat Csikos
+ */
+public final class GradleHoverHelp {
```

Comment:
Nitpick: This is not Gradle specific, so doesn't heed `Gradle` in the name. Maybe name it `HoverText`?